bftengine/src/bftengine/PersistentStorageImp.cpp

```diff
@@ -317,112 +315,104 @@ void PersistentStorageImp::initDescriptorOfLastExecution() {
   size_t actualSize = seqNumData.serializePrePrepareMsg(movablePtr);
   uint32_t itemId = BEGINNING_OF_SEQ_NUM_WINDOW + PRE_PREPARE_MSG + shift;
   Assert(itemId < BEGINNING_OF_CHECK_WINDOW);
-  metadataStorage_->writeInTransaction(itemId, buf, actualSize);
+  metadataStorage_->writeInTransaction(itemId, buf.get(), actualSize);
   Assert(actualSize != 0);
 
-  movablePtr = buf;
+  movablePtr = buf.get();
   actualSize = seqNumData.serializeFullCommitProofMsg(movablePtr);
   itemId = BEGINNING_OF_SEQ_NUM_WINDOW + FULL_COMMIT_PROOF_MSG + shift;
   Assert(itemId < BEGINNING_OF_CHECK_WINDOW);
-  metadataStorage_->writeInTransaction(itemId, buf, actualSize);
+  metadataStorage_->writeInTransaction(itemId, buf.get(), actualSize);
   Assert(actualSize != 0);
 
-  movablePtr = buf;
+  movablePtr = buf.get();
   actualSize = seqNumData.serializePrepareFullMsg(movablePtr);
   itemId = BEGINNING_OF_SEQ_NUM_WINDOW + PRE_PREPARE_FULL_MSG + shift;
   Assert(itemId < BEGINNING_OF_CHECK_WINDOW);
-  metadataStorage_->writeInTransaction(itemId, buf, actualSize);
+  metadataStorage_->writeInTransaction(itemId, buf.get(), actualSize);
   Assert(actualSize != 0);
 
-  movablePtr = buf;
+  movablePtr = buf.get();
   actualSize = seqNumData.serializeCommitFullMsg(movablePtr);
   itemId = BEGINNING_OF_SEQ_NUM_WINDOW + COMMIT_FULL_MSG + shift;
   Assert(itemId < BEGINNING_OF_CHECK_WINDOW);
-  metadataStorage_->writeInTransaction(itemId, buf, actualSize);
+  metadataStorage_->writeInTransaction(itemId, buf.get(), actualSize);
   Assert(actualSize != 0);
 
-  movablePtr = buf;
+  movablePtr = buf.get();
   actualSize = seqNumData.serializeForceCompleted(movablePtr);
   itemId = BEGINNING_OF_SEQ_NUM_WINDOW + FORCE_COMPLETED + shift;
   Assert(itemId < BEGINNING_OF_CHECK_WINDOW);
-  metadataStorage_->writeInTransaction(itemId, buf, actualSize);
+  metadataStorage_->writeInTransaction(itemId, buf.get(), actualSize);
 
-  movablePtr = buf;
+  movablePtr = buf.get();
   actualSize = seqNumData.serializeSlowStarted(movablePtr);
   itemId = BEGINNING_OF_SEQ_NUM_WINDOW + SLOW_STARTED + shift;
   Assert(itemId < BEGINNING_OF_CHECK_WINDOW);
-  metadataStorage_->writeInTransaction(itemId, buf, actualSize);
-  delete[] buf;
-}
-
-void PersistentStorageImp::setSeqNumDataElement(const SeqNum &index, const SharedPtrSeqNumWindow &seqNumWindow) const {
-  setSeqNumDataElement(index, seqNumWindow.get()->getByRealIndex(index));
+  metadataStorage_->writeInTransaction(itemId, buf.get(), actualSize);
 }
 
-void PersistentStorageImp::serializeAndSaveCheckWindow(const SharedPtrCheckWindow &checkWindow) {
-  writeBeginningOfActiveWindow(BEGINNING_OF_CHECK_WINDOW, checkWindow.get()->getBeginningOfActiveWindow());
-  for (uint32_t i = checkWindowFirst_; i < checkWindowLast_; ++i)
-    setCheckDataElement(i, checkWindow);
+void PersistentStorageImp::saveDefaultsInCheckWindow() {
+  writeBeginningOfActiveWindow(BEGINNING_OF_CHECK_WINDOW, checkWindowBeginning_);
+  const CheckData checkData;
```

Comment:
Same as above. Do we need to pass this parameter in, or can it be local to to setCheckDataElement?

Reply:
Explained before.